### ISSUE TITLE
Support comments at the end of header lines

### ISSUE THREAD

**Issue by newlandsvalley** [newlandsvalley/purescript-abc-parser#25]
2.2.5 Comments and remarks
A percent symbol (%) will cause the remainder of any input line to be ignored. It can be used to add a comment to the end of an abc line or as a comment line in its own right. (To get a percent symbol, type \% - see text strings.)

Alternatively, you can use the syntax [r:remark] to write a remark in the middle of a line of music.

Example:

````
|:DEF FED| % this is an end of line comment
% this is a comment line
DEF [r:and this is a remark] FED:|
```

**Comment by newlandsvalley** [comment]
I've corrected this for the X header only.  This change is more trouble than it's worth at the moment.  The main issue is that headers can appear either free-standing or inline. Many free-standing only headers cope with terminating comments because they read till the end of the line and discard the contents. However headers that can occur in both places don't do that.  

Maybe there's a shorthand way of implementing this by recognising a ```%``` character at a higher level in the parse.  This seems quite disruptive to the overall parser design.  I'll sit on this till I've thought of something.

I would doubt if anyone would ever put a comment inside an inline header and it seems as if the spec has been designed without considering this as a possibility.  

Or perhaps we should redefine the EOL to include comments?  That's perhaps a better idea.

**Comment by newlandsvalley** [comment]
OK - the EOL approach worked.  Fixed by 11ebf6a3b10a2eaab71fdcd29c297ac13493902a.
